pkg/component/etcd/statefulset/statefulset.go

```diff
@@ -72,21 +72,22 @@ func (c *component) Deploy(ctx context.Context) error {
 		sts = c.emptyStatefulset()
 	}
 
-	if sts.Generation > 1 && sts.Spec.ServiceName != c.values.PeerServiceName {
-		// Earlier clusters referred to the client service in `sts.Spec.ServiceName` which must be changed
-		// when a multi-node cluster is used, see https://github.com/gardener/etcd-druid/pull/293.
-		if clusterScaledUpToMultiNode(c.values) {
-			deleteAndWait := gardenercomponent.OpDestroyAndWait(c)
-			if err := deleteAndWait.Destroy(ctx); err != nil {
-				return err
-			}
-			sts = c.emptyStatefulset()
+	if sts.Generation > 1 && clusterScaledUpToMultiNode(c.values) && immutableFieldUpdate(sts, c.values) {
+		// Several immutable fields must be reset for the multi-node use-case.
+		deleteAndWait := gardenercomponent.OpDestroyAndWait(c)
+		if err := deleteAndWait.Destroy(ctx); err != nil {
+			return err
 		}
+		sts = c.emptyStatefulset()
 	}
 
 	return c.syncStatefulset(ctx, sts)
 }
 
+func immutableFieldUpdate(sts *appsv1.StatefulSet, val Values) bool {
+	return sts.Spec.ServiceName != val.PeerServiceName || sts.Spec.PodManagementPolicy != appsv1.ParallelPodManagement
+}
```

Comment:
Should we generalise this and let it return true if any immutable field in the sts is being updated? (any field apart from `replicas`, `template`, `updateStrategy`, or `minReadySeconds`)
wdyt?

Reply:
Good question. FMPOV we should not do this in general because such a deletion should remain an exception, i.e. it should be a conscious decision of Druid under which circumstances the deletion and thus an interruption is acceptable, see https://github.com/gardener/etcd-druid/issues/404#issuecomment-1215129643. WDYT?

Reply:
AFAIK immutability is currently only enforced via webhooks. There is no way currently to know by just looking at the spec which fields are immutable. The [proposal](https://github.com/kubernetes/enhancements/pull/1265) for that is merged but i am unaware if the implementation has been. Anything generic will require this support, right?

Reply:
> AFAIK immutability is currently only enforced via webhooks.

I don't see any issues with that. We do this for provider extensions in Gardener as well. I think it's more dangerous to just allow a field to be changed, delete the sts by default and thus cause a unexpected disruption.

Reply:
Yes there is nothing wrong in using a webhook as that is the only way to enforce it. My comment was for:
>Should we generalise this and let it return true if any immutable field in the sts is being updated?

To  enforce of all immutable fields in a spec generically you need to be aware of all immutable fields. I was merely pointing to the fact that this is not possible for any spec today as immutable field identification is not yet part of a spec but is only coded in the webhook.

Reply:
Fair enough, thank you for pointing this out 
Since I see we're all in agreement, I will resolve this conversation :)